core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -136,7 +137,7 @@ public PartitionSpec spec() {
     return spec;
   }
 
-  public ManifestReader<F> select(Collection<String> newColumns) {
+  public ManifestReader<F> select(List<String> newColumns) {
```

Comment:
Why was it necessary to change this to `List` instead of `Collection`? That's not a binary-compatible change. Couldn't we just make a copy of the collection if we need the field to be a list? I'm also not sure why `columns` can't be a `Collection`.

Reply:
I think I modified this when trying to address [this comment](https://github.com/apache/iceberg/pull/1820#discussion_r530610375), as I noticed that all usage of `columns` could be done via list so I directly changed the type to avoid the list copying, without thinking about backward compatibility of this method. I guess your original suggestion was to move the copying from `ManifestGroup` to `ManifestReader`, but I misinterpreted it to get rid of the list copy completely?